### DISCUSSION TITLE
Workaround for vulkan2 error

### DISCUSSION THREAD

**Discussion by keiohta** [haosulab/SAPIEN#86] [Q&A] [answered]
Hi, I encountered an error on my GPU cluster and cannot run scripts. My SAPIEN version is 1.1.1, and the following are the contents of the script (basically just trying to create a renderer object) and the error.

```
$ cat temp.py 
import sapien.core as sapien
engine = sapien.Engine()
renderer = sapien.VulkanRenderer(offscreen_only=True)

$ srun -X --gres gpu:1 python temp.py
[2022-08-23 11:21:35.356] [svulkan2] [error] Some required Vulkan extension is not present. You may not use the renderer to render, however, CPU resources will be still available.
```

I found a workaround in [this issue](https://github.com/haosulab/ManiSkill-Learn/issues/1#issuecomment-892338182), but I found that a GPU node does not even have `/usr/share/vulkan`. However, since the GPU cluster I am using is a big shared one, so it would be hard to install `vulkan` there. So... can you give me some advice to run SAPIEN in my environment?

Thanks,
Kei

**Comment by fbxiang** [comment, marked as answer]
In fact, the ICD file does not need to live in /usr/share/vulkan. If your graphics driver is installed properly, then you can just copy over that ICD file from some other system, put it anywhere, and set the environment variable accordingly.
If that does not work, then I can only imagine your system does not have a proper graphics driver.

**Reply by keiohta** [reply]
Thanks for the reply and sorry for being late. It seems the reason can be a bit old driver (`440.64.00`). I checked SAPIEN works on other setups with driver versions: `455.23.05` and `470.63.01`. Unfortunately, I don't have permission to update the driver... so it would be great if you can share some workarounds.

**Reply by fbxiang** [reply]
Unfortunately driver version > 470 is a hard requirement, as I believe proper Vulkan support only comes sometime between 460-470. I would even recommend upgrading to > 500 if possible, as 470 is still missing some key features, but 470 should work fine as long as you do not use ray tracing.

**Reply by keiohta** [reply]
Oh wow... thanks for letting me know that! I then give up using the GPU clusters and find other machine resources. Thanks!
